src/ast.cpp

```diff
@@ -13,7 +13,7 @@
 namespace {
 
 /// \return tokenizer flags corresponding to parse tree flags.
-static tok_flags_t tokenizer_flags_from_parse_flags(parse_tree_flags_t flags) {
+tok_flags_t tokenizer_flags_from_parse_flags(parse_tree_flags_t flags) {
```

Comment:
My bad, I didn't realize that these were in an anonymous namespace.

I would prefer if we put the `namespace {}` only around the class (`token_stream_t`) and keep the functions static to make the fact harder to miss, see also [LLVM's rationale on this](http://llvm.org/docs/CodingStandards.html#anonymous-namespaces).

We also have other classes defined in cpp files, like `completer_t`, they should have internal linkage although I don't know if it makes a difference.

Reply:
I took out all those functions from the namespace. errors.

```
src/ast.cpp:19:41: error: use of undeclared identifier 'tokenizer_flags_from_parse_flags'
        : src_(src), tok_(src_.c_str(), tokenizer_flags_from_parse_flags(flags)) {}
                                        ^
src/ast.cpp:82:30: error: use of undeclared identifier 'parse_token_type_from_tokenizer_token'
        parse_token_t result{parse_token_type_from_tokenizer_token(token.type)};
                             ^
src/ast.cpp:84:26: error: use of undeclared identifier 'keyword_for_token'
        result.keyword = keyword_for_token(token.type, text);

```

Reply:
Hmm if I simply move the namespace down on master, it builds fine:

```diff
diff --git a/src/ast.cpp b/src/ast.cpp
index c8749df0f..3e20c0e99 100644
--- a/src/ast.cpp
+++ b/src/ast.cpp
@@ -11,6 +11,4 @@
 #include "wutil.h"
 
-namespace {
-
 /// \return tokenizer flags corresponding to parse tree flags.
 static tok_flags_t tokenizer_flags_from_parse_flags(parse_tree_flags_t flags) {
@@ -102,4 +100,5 @@ static parse_token_type_t parse_token_type_from_tokenizer_token(
 }
 
+namespace {
 /// A token stream generates a sequence of parser tokens, permitting arbitrary lookahead.
 class token_stream_t {
@@ -215,5 +214,4 @@ class token_stream_t {
     wcstring storage_;
 };
-
 }  // namespace
```

Reply:
Fixed.